spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'posts', type: :request do
+  describe 'GET /posts' do
+    it 'works! returns correct status code' do
+      get user_posts_path(user_id: 1)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'works!  renders index view correctly' do
+      get user_posts_path(user_id: 1)
+      expect(response).to render_template(:index)
+    end
+
+    it 'works! reponse body includes <h1>This is a list of posts<h1>' do
+      get user_posts_path(user_id: 1)
+      expect(response.body).to include('<h1>This is a list of posts<h1>')
+    end
+  end
+end
+
+RSpec.describe 'posts/:id', type: :request do
```

Comment:
- [ ] I would recommend removing this `describe` from here to follow the DRY and rails principle.